analyzers/nodejs/integration_test.go

```diff
@@ -0,0 +1,203 @@
+	waitGroup.Add(len(projects))
+
+	for _, project := range projects {
+		go func(proj testUtils.ProjectFixture) {
+			defer waitGroup.Done()
+			println("installing " + proj.Name)
+			projectDir := filepath.Join(testDir, proj.Name)
+
+			stdOut, errOut, err := exec.Run(exec.Cmd{
+				Name:    "npm",
+				Argv:    []string{"install", "--production"},
+				Dir:     projectDir,
+				WithEnv: proj.Env,
+				Command: "npm",
+			})
+			if err != nil {
+				println(errOut)
+				println("failed to run npm install on " + proj.Name)
+			}
+
+			println(stdOut)
+
+			// save time on local
+			ymlAlreadyExists, err := files.Exists(filepath.Join(projectDir, ".fossa.yml"))
+			if err != nil {
+				panic(err)
+			}
+			if ymlAlreadyExists {
+				return
+			}
+
+			println("starting fossa init for " + proj.Name)
+			stdOut, err = testUtils.FossaInit(projectDir)
+			if err != nil {
+				println("failed to run fossa init on " + proj.Name)
+				println(err.Error())
+				panic(err)
+			}
+			println(stdOut)
+			println("completed fossa init for " + proj.Name)
+		}(project)
+	}
+
+	waitGroup.Wait()
+
+	return nil
+}
+
+var projects = []testUtils.ProjectFixture{
+	testUtils.ProjectFixture{
+		Name:   "puppeteer",
+		URL:    "https://github.com/GoogleChrome/puppeteer",
+		Commit: "b97bddf8e5750d20c6ba82392eebe2a3fd2dd218",
+		Env: map[string]string{
+			"PUPPETEER_SKIP_CHROMIUM_DOWNLOAD": "1",
+		},
+	},
+	testUtils.ProjectFixture{
+		Name:   "fakerjs",
+		URL:    "https://github.com/Marak/faker.js",
+		Commit: "3a4bb358614c1e1f5d73f4df45c13a1a7aa013d7",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "fastify",
+		URL:    "https://github.com/fastify/fastify",
+		Commit: "1b16a4c5e381f9292d3ac2c327c3bda4bd277408",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "nest",
+		URL:    "https://github.com/nestjs/nest",
+		Commit: "ce498e86150f7de4a260f0c393d47ec4cc920ea1",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "ohm",
+		URL:    "https://github.com/harc/ohm",
+		Commit: "8202eff3723cfa26522134e7b003cf31ab5de445",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "express",
+		URL:    "https://github.com/expressjs/express",
+		Commit: "b4eb1f59d39d801d7365c86b04500f16faeb0b1c",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "standard",
+		URL:    "https://github.com/standard/standard",
+		Commit: "bc02256fa2c03632e657248483c55a752e63e724",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "sodium-encryption",
+		URL:    "https://github.com/mafintosh/sodium-encryption",
+		Commit: "42a7cba0f97718157e8c7a386ef94ba31e16837a",
+		Env:    map[string]string{},
+	},
+	testUtils.ProjectFixture{
+		Name:   "request",
+		URL:    "https://github.com/request/request",
+		Commit: "8162961dfdb73dc35a5a4bfeefb858c2ed2ccbb7",
+		Env:    map[string]string{},
+	},
+}
+
+func cleanUp(dir string) {
```

Comment:
I'm still on the fence as to whether or not we should cleanup the fixtures after the run. It's not necessary for CI and it's also not necessary for local dev; it will likely save time. If you need a new run locally, you can just delete the fixtures manually

Reply:
We should not clean up fixtures. Instead, they should be left on disk (or saved in cache in CI) and we should verify their integrity before running (for git repositories, this is probably as simple checking that `git status` is clean and the revision hash is correct).